File: C++/comparing.cpp
#include <iostream>
#include <string>

#define YES 1
#define NO -1
#define EQ 0

void input(std::string* S);
void output(std::string S);
int check(std::string S);
bool isVowel(char S);

int main(){
    std::string S;
    input(&S);
    output(S);
}

void input(std::string* S){
    getline(std::cin, *S);
}

void output(std::string S){
    int result = check(S);
    std::cout << (result == YES ? "Да" : (result == NO ? "Нет" : "Одинаково"));
}

int check(std::string S){
    int numOfVowels = 0;
    int numOfConsonants = 0;
    int numOfOthers = 0;
    for (char s : S){
        isVowel(s) ? numOfVowels++ : (s != ' ' ? numOfConsonants++ : numOfOthers++); 
    }
    return numOfVowels > numOfConsonants ? YES : (numOfVowels < numOfConsonants ? NO : EQ);
}

bool isVowel(char s){
    std::string vowels = "aeiouyAEIOUY";
    bool isVow = false;
    for (char v : vowels){
        if (s == v) isVow = true;
    }
    return isVow;
}

File: Rust/comparing.rs
use std::io;

const YES: i32 = 1;
const NO: i32 = -1;
const EQ: i32 = 0;

fn main() {
    let s = input();
    output(&s);
}

fn input() -> String {
    let mut s = String::new();
    let _ = io::stdin().read_line(&mut s);
    s.to_string()
}

fn output(s: &str) {
    let result = check(s);
    match result {
        YES => print!("Да"),
        NO => print!("Нет"),
        _ => print!("Одинаково"),
    }
}

fn check(s: &str) -> i32 {
    let mut num_of_vowels = 0;
    let mut num_of_consonants = 0;
    
    for c in s.chars() {
        if is_vowel(c) {
            num_of_vowels += 1;
        } else if c != ' ' {
            num_of_consonants += 1;
        }
    }
    
    if num_of_vowels > num_of_consonants {
        YES
    } else if num_of_vowels < num_of_consonants {
        NO
    } else {
        EQ
    }
}

fn is_vowel(c: char) -> bool {
    match c.to_ascii_lowercase() {
        'a' | 'e' | 'i' | 'o' | 'u' | 'y' => true,
        _ => false,
    }
}